pkg/controller/velerobackup/hiveobject.go

```diff
@@ -0,0 +1,113 @@
+package velerobackup
+
+import (
+	"fmt"
+
+	controllerutils "github.com/openshift/hive/pkg/controller/utils"
+	log "github.com/sirupsen/logrus"
+	meta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1alpha1"
+)
+
+const (
+	errChecksum = "HIVE_CHECKSUM_ERR_97A29D08"
+)
+
+type hiveObject struct {
+	checksum string
+	object   runtime.Object
+	logger   log.FieldLogger
+}
+
+func newHiveObject(object runtime.Object, logger log.FieldLogger) (*hiveObject, error) {
+	switch t := object.(type) {
+	case *hivev1.ClusterDeployment:
+		checksum := calculateChecksum(t.ObjectMeta, t.Spec, controllerutils.GetChecksumOfObjects, logger)
+		return &hiveObject{
+			checksum: checksum,
+			object:   object,
+			logger:   logger,
+		}, nil
+	case *hivev1.SyncSet:
+		checksum := calculateChecksum(t.ObjectMeta, t.Spec, controllerutils.GetChecksumOfObjects, logger)
+		return &hiveObject{
+			checksum: checksum,
+			object:   object,
+			logger:   logger,
+		}, nil
+	case nil:
+		return nil, fmt.Errorf("Can't convert nil to a HiveObject")
+	}
+
+	return nil, fmt.Errorf("Unknown Type: %T", object)
+}
+
+func (h *hiveObject) hasChanged() bool {
+	if h.checksum == errChecksum {
+		// The checksum function errored. In this case we assume the object has NOT changed.
```

Comment:
I think we want to treat `errChecksum` identically to all other checksums. If the calculated checksum and the annotation checksum are different, then the object has changed, regardless of whether on of those checksums is `errChecksum`.